File: src/lib/tauri/commands/shell.ts
/** Shell 命令调用封装 */

import { invoke } from "@tauri-apps/api/core";

export interface CommandOutput {
  code: number;
  stdout: string;
  stderr: string;
}

export const shellCommands = {
  run: (program: string, args: string[] = []): Promise<CommandOutput> =>
    invoke<CommandOutput>("shell_run", { program, args }),
};

File: src-tauri/src/commands/shell.rs
//! Shell 相关 command

use crate::error::AppResult;
use crate::models::CommandOutput;
use crate::services::shell_service;

#[tauri::command]
pub fn shell_run(program: String, args: Vec<String>) -> AppResult<CommandOutput> {
    shell_service::run(&program, &args)
}
